Classes/Domain/NodeCreation/PropertiesHandler.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace Flowpack\NodeTemplates\Domain\NodeCreation;
+
+use Flowpack\NodeTemplates\Domain\ExceptionHandling\CaughtException;
+use Flowpack\NodeTemplates\Domain\ExceptionHandling\CaughtExceptions;
+use Neos\ContentRepository\Domain\Model\NodeInterface;
+use Neos\ContentRepository\Domain\Model\NodeType;
+use Neos\ContentRepository\Domain\Service\Context;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Property\Exception as PropertyMappingException;
+use Neos\Flow\Property\PropertyMapper;
+use Neos\Flow\Property\PropertyMappingConfiguration;
+
+/**
+ * @Flow\Proxy(false)
+ */
+class PropertiesHandler
+{
+    private Context $subgraph;
+
+    private PropertyMapper $propertyMapper;
+
+    public function __construct(Context $subgraph, PropertyMapper $propertyMapper)
+    {
+        $this->subgraph = $subgraph;
+        $this->propertyMapper = $propertyMapper;
+    }
+
+    public function createdFromArrayByTypeDeclaration(array $propertiesAndReferences, NodeType $nodeType): Properties
+    {
+        $references = [];
+        $properties = [];
+        foreach ($propertiesAndReferences as $propertyName => $propertyValue) {
+            // TODO: remove the next line to initialise the nodeType, once https://github.com/neos/neos-development-collection/issues/4333 is fixed
+            $nodeType->getFullConfiguration();
+            $declaration = $nodeType->getPropertyType($propertyName);
+            if ($declaration === 'reference' || $declaration === 'references') {
+                $references[$propertyName] = $propertyValue;
+                continue;
+            }
+            $properties[$propertyName] = $propertyValue;
+        }
+        return new Properties($properties, $references, $nodeType);
+    }
+
+    /**
+     * A few checks are run against the properties before they are applied on the node.
+     *
+     * 1. It is checked, that only properties will be set, that were declared in the NodeType
+     *
+     * 2. It is checked, that the property value is assignable to the property type.
+     *    In case the type is class or an array of classes, the property mapper will be used map the given type to it. If it doesn't succeed, we will log an error.
+     */
+    public function requireValidProperties(Properties $properties, CaughtExceptions $caughtExceptions): array
```

Comment:
```suggestion
    public function getValidatedProperties(Properties $properties, CaughtExceptions $caughtExceptions): array
```

Reply:
Or `getValidProperties`?

Reply:
the `require*` pattern is inspired by the new ESCR but i agree that i found it odd at first to, so ill revert this.

How about `PropertiesProcessor::processAndValidateProperties`